src/linked_list.js

```diff
@@ -0,0 +1,172 @@
+'use strict'
+
+class Node {
+  constructor(data, next) {
+    this.data = data
+    this.next = next
+  }
+}
+
+export default class LinkedList {
+  constructor() {
+    this.list = null
+    this.size = 0
+  }
+
+  getHeadNode() {
+    return this.list
+  }
+
+  getTailNode() {
+    let temp = this.list
+    while (temp.next) {
+      temp = temp.next
+    }
+    return temp
+  }
+
+  contains(value) {
+    let status = false
+    let temp = this.list
+    while (temp) {
+      if (temp.data === value) {
+        status = true
+      }
+      temp = temp.next
+    }
+    return status
+  }
+
+  find(value) {
+    let temp = this.list
+    while (temp) {
+      if (temp.data === value) {
+        return temp
+      }
+      temp = temp.next
+    }
+    return -1
+  }
+
+  insert(value) {
+    if (!this.list) {
+      this.list = new Node(value, null)
+      this.size++
+      return
```

Comment:
this return seems a little hacky, I think this function would be better set up as a single if else statement.